fv3core/fv3core/initialization/baroclinic.py

```diff
@@ -0,0 +1,522 @@
+        jslice=slice(0, ny),
+        jslice_grid=slice(1, ny + 1),
+        axis=1,
+    )
+
+    initialize_zonal_wind(
+        u,
+        eta,
+        eta_v,
+        lon,
+        lat,
+        east_grid_vector_component=ee1,
+        center_grid_vector_component=es1,
+        islice=slice(0, nx),
+        islice_grid=slice(1, nx + 1),
+        jslice=slice(0, ny + 1),
+        jslice_grid=slice(0, ny + 1),
+        axis=0,
+    )
+
+    # Compute cell lats in the midpoint of each cell edge
+    lat2, lat3, lat4, lat5 = compute_grid_edge_midpoint_latitude_components(lon, lat)
+    slice_3d = (slice(0, nx), slice(0, ny), slice(None))
+    slice_2d = (slice(0, nx), slice(0, ny))
+    # initialize temperature
+    t_mean = jablo_init.horizontally_averaged_temperature(eta)
+    pt[slice_3d] = cell_average_nine_components(
+        jablo_init.temperature,
+        [eta, eta_v, t_mean],
+        lat,
+        lat_agrid,
+        lat2,
+        lat3,
+        lat4,
+        lat5,
+        slice_2d,
+    )
+
+    # initialize surface geopotential
+    phis[slice_2d] = cell_average_nine_components(
+        jablo_init.surface_geopotential_perturbation,
+        [],
+        lat,
+        lat_agrid,
+        lat2,
+        lat3,
+        lat4,
+        lat5,
+        slice_2d,
+    )
+
+    if not hydrostatic:
+        # vertical velocity is set to 0 for nonhydrostatic setups
+        w[slice_3d] = 0.0
+        delz[:nx, :ny, :-1] = initialize_delz(pt[slice_3d], peln[slice_3d])
+
+    if not adiabatic:
+        qvapor[:nx, :ny, :-1] = jablo_init.specific_humidity(
+            delp[slice_3d], peln[slice_3d], lat_agrid[slice_2d]
+        )
+        pt[slice_3d] = moisture_adjusted_temperature(pt[slice_3d], qvapor[slice_3d])
+
+
+def initialize_pkz_moist(delp, pt, qvapor, delz):
+    return np.exp(
+        constants.KAPPA
+        * np.log(
+            constants.RDG
+            * delp[:, :, :-1]
+            * pt[:, :, :-1]
+            * (1.0 + constants.ZVIR * qvapor[:, :, :-1])
+            / delz[:, :, :-1]
+        )
+    )
+
+
+def initialize_pkz_dry(delp, pt, delz):
+    return np.exp(
+        constants.KAPPA
+        * np.log(constants.RDG * delp[:, :, :-1] * pt[:, :, :-1] / delz[:, :, :-1])
+    )
+
+
+def fix_top_log_edge_pressure(peln, ptop):
+    if ptop < ptop_min:
+        ak1 = (constants.KAPPA + 1.0) / constants.KAPPA
+        peln[:, :, 0] = peln[:, :, 1] - ak1
+    else:
+        peln[:, :, 0] = np.log(ptop)
+
+
+def p_var(
+    delp,
+    delz,
+    pt,
+    ps,
+    qvapor,
+    pe,
+    peln,
+    pkz,
+    ptop,
+    moist_phys,
+    make_nh,
+    hydrostatic=False,
+    adjust_dry_mass=False,
+):
+    """
+    Computes auxiliary pressure variables for a hydrostatic state.
+
+    The Fortran code also recomputes some more pressure variables,
+    pe, pk, but since these are already done in setup_pressure_fields
+    we don't duplicate them here
+    """
+    assert not adjust_dry_mass
+    assert not hydrostatic
```

Comment:
Does it make sense for us to just not implement these as keyword arguments? I don't think we have any plans to implement these, the code itself is easier to understand than the rest of the dycore in terms of if we were to implement a new option, and if we did implement a new option we'd probably do it using new functions rather than if flags?